### PR TITLE
INREL-3764 new article trigger

### PR DESCRIPTION
**Pull request by Nicozilla**
## [INREL-3764](https://jira.burda.com/browse/INREL-3764) 
- new article_footer_blocks_common in common template. So we can add dynamically the next-article-info-layer for specific platform (InStyle next article scroll logic when you see the preloader

## How to test:
Scroll fast to the next article. Now you see the preloader. After the preloading is done it's scrolling automatically to the next article.

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](http://programmer.97things.oreilly.com/wiki/index.php/The_Boy_Scout_Rule)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

File: js/infinite/libs/shortcuts/infinite_custom.js [hunk 1]
```diff
@@ -86,6 +86,8 @@
     var $newMore = $data.find(this.options.more);
     var $items = $data.find(this.options.items);
 
+    this.options.onAfterPrePageLoad($items);
```
Comment (saviomuc) [inline js/infinite/libs/shortcuts/infinite_custom.js]: 1. What is the purpose of this line?

2. Naming: `onAfterPrePageLoad` sounds confusing to me. Further down it is a reference to `onAfterPrePageLoad: _.bind(this.onAfterPreLoad, this),`. I think we should have consistent names to make it as easy as possible to understand.

2. Usage: Can you also clarify when the event triggers?

File: js/infinite/views/base/base-feed-view.js [hunk 2]
```diff
@@ -9,11 +9,15 @@
     preloader: null,
     infinite: null,
     initFeed: true,
+    scrollToAppendedElement: false,
     onBeforeLoadCallback: $.noop,
+    onAfterPrePageLoad: $.noop,
     onAfterLoadCallback: $.noop,
+    naviHeight: 56,
```
Comment (saviomuc) [inline js/infinite/views/base/base-feed-view.js]: This is a magic number. Where does 56 come from? Is there a reason it has to be in base-feed-view. If it's mapped to the height of the navigation element how can we make it reusable across all other pages?

File: js/infinite/views/base/base-feed-view.js [hunk 3]
```diff
@@ -9,11 +9,15 @@
...
     initialize: function (pOptions) {
       BaseDynamicView.prototype.initialize.call(this, pOptions);
 
+      this.naviHeight = document.querySelector('#menu-main-navigation').offsetHeight || this.naviHeight;
```
Comment (saviomuc) [inline js/infinite/views/base/base-feed-view.js]: In that case wouldn't it make more sense to use getElementByID? Also I am not sure if using an ID is a good idea. The ID could potentially change (in template and CSS). A solution which is a little bit more stable would be a class with some sort of prefix like `js-main-menu-navigation`

File: js/infinite/views/base/base-feed-view.js [hunk 4]
```diff
@@ -31,6 +35,7 @@
         context: this.context[0],
         element: this.$el,
         onBeforePageLoad: _.bind(this.onBeforeLoad, this),
+        onAfterPrePageLoad: _.bind(this.onAfterPreLoad, this),
```
Comment (saviomuc) [inline js/infinite/views/base/base-feed-view.js]: See comment above at https://github.com/BurdaMagazinOrg/infinite-theme/pull/134/files#diff-ba53da8dd514fd24e59094b00fe41f91R89

File: js/infinite/views/base/base-feed-view.js [hunk 5]
```diff
@@ -43,15 +48,35 @@
       this.preloader = new SpinnerCubeView({el: this.lastInfiniteItem});
       this.onBeforeLoadCallback(this.lastInfiniteItem);
     },
+    onAfterPreLoad: function (pItem) {
+      /**
```
Comment (saviomuc) [inline js/infinite/views/base/base-feed-view.js]: Please use single line comments if the comment is on one line.

File: js/infinite/views/base/base-feed-view.js [hunk 6]
```diff
@@ -43,15 +48,35 @@
...
+       * scroll to new appended article if someone scroll to the bottom and saw the preloader
+       */
+      if (((window.scrollY + window.innerHeight) === document.body.clientHeight) && $('body').hasClass('page-article')) {
```
Comment (saviomuc) [inline js/infinite/views/base/base-feed-view.js]: Can we use something more descriptive? Like `var atBottomOfPage = (window.scrollY + window.innerHeight) === document.body.clientHeight`

Comment (saviomuc) [reply js/infinite/views/base/base-feed-view.js]: Also not sure why we need the condition `$('body').hasClass('page-article')`. I think you are mixing two different contexts here (scroll position and execution scope). execution scope should be handled somewhere.

File: js/infinite/views/base/base-feed-view.js [hunk 7]
```diff
@@ -43,15 +48,35 @@
...
+        this.scrollToAppendedElement = true;
```
Comment (saviomuc) [inline js/infinite/views/base/base-feed-view.js]: Can we use the model here and use `listenTo`? Or is there a reason not to use it?

File: js/infinite/views/base/base-feed-view.js [hunk 8]
```diff
@@ -43,15 +48,35 @@
...
+      }
+    },
     onAfterLoad: function (pItem) {
       //console.log("loading done", " View >> ", this.id || this.$el.attr('id') || this.$el.attr('class'));
```
Comment (saviomuc) [inline js/infinite/views/base/base-feed-view.js]: Please remove unused line

File: js/infinite/views/base/base-feed-view.js [hunk 9]
```diff
@@ -43,15 +48,35 @@
...
+      var tmpElement = pItem[0];
```
Comment (saviomuc) [inline js/infinite/views/base/base-feed-view.js]: What is the meaning of `tmpElement`? What does it contain? What is the meaning of `pItem`? Can we use more specific variable names?

File: js/infinite/views/base/base-feed-view.js [hunk 10]
```diff
@@ -43,15 +48,35 @@
...
+        $('html, body').animate({scrollTop: (tmpElement.offsetTop - this.naviHeight)}, {
+          duration: 500,
+          easing: 'easeInOutCubic'
+        });
+
+        this.scrollToAppendedElement = false;
+      }
+
+      this.onAfterLoadCallback($(tmpElement));
+      this.parseInfiniteView($(tmpElement), {modelList: this.model, initialDOMItem: false}); //delegateElements: true,
```
Comment (saviomuc) [inline js/infinite/views/base/base-feed-view.js]: Please remove comment at end

**Review by saviomuc — CHANGES_REQUESTED** [review]
Code seems to work. Nitpicks / Questions regarding naming and structure.

**Review by saviomuc — CHANGES_REQUESTED** [review]
Questions regarding code usage, structure, naming and scope.
Needs rework regarding naming, unused code and structure.

File: js/infinite/libs/shortcuts/infinite_custom.js (hunk 1 above)
Comment (Nicozilla) [reply js/infinite/libs/shortcuts/infinite_custom.js]: removed - not necessary

File: js/infinite/views/base/base-feed-view.js (hunk 3 above)
Comment (Nicozilla) [reply js/infinite/views/base/base-feed-view.js]: getElementByID -> good idea

File: js/infinite/views/base/base-feed-view.js (hunk 2 above)
Comment (Nicozilla) [reply js/infinite/views/base/base-feed-view.js]: we set it already dynamically

File: js/infinite/views/base/base-feed-view.js (hunk 6 above)
Comment (Nicozilla) [reply js/infinite/views/base/base-feed-view.js]: It should only be scrolled on articles.

File: js/infinite/views/base/base-feed-view.js (hunk 9 above)
Comment (Nicozilla) [reply js/infinite/views/base/base-feed-view.js]: Changed. Get jQueryElement -> $appendedElement
```javascript 
      onAfterPageLoad: function ($appendedElement) {
      var appendedElement = $appendedElement[0];
```

File: js/infinite/views/base/base-feed-view.js (hunk 7 above)
Comment (Nicozilla) [reply js/infinite/views/base/base-feed-view.js]: On this time and position not practicable. We want the check before the content is appended. When the content is appended and the flag is true - scroll to next article.

File: js/infinite/views/base/base-feed-view.js [hunk 11]
```diff
@@ -30,31 +31,46 @@
...
       this.preloader = new SpinnerCubeView({el: this.lastInfiniteItem});
-      this.onBeforeLoadCallback(this.lastInfiniteItem);
     },
-    onAfterLoad: function (pItem) {
-      //console.log("loading done", " View >> ", this.id || this.$el.attr('id') || this.$el.attr('class'));
+    preAppendCallback: function (pItem) {
+      //scroll to new appended article if someone scroll to the bottom and saw the preloader
+      var atBottomOfPageCheck = ((window.scrollY + window.innerHeight) === document.body.clientHeight);
+      if (atBottomOfPageCheck) {
+        this.atBottomOfPage = true;
```
Comment (saviomuc) [inline js/infinite/views/base/base-feed-view.js:51]: Just a note, you could also do:

`this.atBottomOfPage = ((window.scrollY + window.innerHeight) === document.body.clientHeight)`

As the assignment key (atBottomOfPage) is already clear here.

**Review by saviomuc — APPROVED** [review]
One note.

LGTM :shipit:

**Pull request merged**